scripts/wrappers/enable.py

```diff
@@ -1,20 +1,43 @@
-#!/usr/bin/python3
+#!/usr/bin/env python3
 
-from common.utils import get_available_addons, get_current_arch
+import click
 
+from common.utils import (
+    ensure_started,
+    exit_if_no_permission,
+    is_cluster_locked,
+    print_help,
+    wait_for_ready,
+    xable,
+)
+from status import get_available_addons, get_current_arch, get_status
 
-def print_console(addons):
-    print("Available Addons:")
-    for addon in addons:
-        print("{:>1} {:<20} # {}".format("", addon["name"], addon["description"]))
 
+@click.command(context_settings={'ignore_unknown_options': True})
+@click.argument('addons', nargs=-1, required=True)
+def enable(addons):
+    """Enables a MicroK8s addon.
 
-def show_help():
-    print("Usage: microk8s enable ADDON...")
-    print("Enable one or more ADDON included with microk8s")
-    print("Example: microk8s enable dns storage")
+    For a list of available addons, run `microk8s status`.
 
+    To see help for individual addons, run:
 
-available_addons = get_available_addons(get_current_arch())
-show_help()
-print_console(available_addons)
+        microk8s enable ADDON -- --help
+    """
+
+    if print_help(addons):
+        return
+
+    is_cluster_locked()
+    exit_if_no_permission()
+    ensure_started()
+    wait_for_ready()
```

Comment:
The behavior we had until now is to wait for 30 seconds not wait forever. This is because it is better to say, "tried to enable the addon and failed" than to hang.